### PR TITLE
refactor(meta): track finished create mv job in tracker

### PR DESCRIPTION
**Pull request by wenym1**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Previously we track the finished create mv job in `CheckpointControl`. However, as the `CheckpointControl` is exclusively owned by the global barrier manager worker loop, we are unable, or complicated, to write code that spawn a task to complete(call `finish_job`) the finished create mv job. In this PR we will move to track the finished create mv job to the `CreateMviewProgressTracker`, which is shared by `Arc<Mutex<...>>`. Moreover, `CheckpointControl` is intended more to handle barrier state, and is less relevant to track create mv related information, and therefore this makes our code more clear.

Another change included in this PR is: previously in `complete_barrier`, if we are to cancel a create mv command, we will first get the command to be cancelled from the tracker, and then we try to remove the command from the queue in `CheckpointControl`. However, this is unnecessary, because the command to remove must not exist in the queue. A previous command can only be added to the tracker within the `complete_barrier` method, after which the command is removed from the queue in `CheckpointControl`, and therefore, when handling a later command to cancel this previous command, the previous command must not lay in the queue of checkpoint control. For this reason, the `cancel_command` of the `Checkpoint` must be no-op, and in this PR, we remove the `cancel_command` of the `CheckpointControl`. We only clear the tracking information of the cancelled command in the tracker.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/meta/src/barrier/mod.rs [hunk 1]
```diff
@@ -196,76 +196,34 @@ struct CheckpointControl {
...
-
-    fn cancel_command(&mut self, cancelled_job: TrackingJob) {
-        if let TrackingJob::New(cancelled_command) = cancelled_job {
-            if let Some(index) = self.command_ctx_queue.iter().position(|x| {
-                x.command_ctx.prev_epoch.value() == cancelled_command.context.prev_epoch.value()
-            }) {
-                self.command_ctx_queue.remove(index);
-            }
-        } else {
-            // Recovered jobs do not need to be cancelled since only `RUNNING` actors will get recovered.
```
Comment (kwannoel) [inline src/meta/src/barrier/mod.rs:244]: Logic moved into `CreateMviewProgressTracker`.

File: src/meta/src/barrier/mod.rs [hunk 2]
```diff
@@ -874,21 +824,16 @@ impl GlobalBarrierManager {
...
+                    tracker.stash_command_to_finish(command);
                 }
 
-                if let Some(command) = cancelled_command {
-                    self.checkpoint_control.cancel_command(command);
-                } else if let Some(table_id) = node.command_ctx.table_to_cancel() {
+                if let Some(table_id) = node.command_ctx.table_to_cancel() {
                     // the cancelled command is possibly stashed in `finished_commands` and waiting
                     // for checkpoint, we should also clear it.
-                    self.checkpoint_control.cancel_stashed_command(table_id);
```
Comment (kwannoel) [inline src/meta/src/barrier/mod.rs:885]: Hmm what about this part. The `finished_commands` could still contain the command to be cancelled.

Comment (wenym1) [reply src/meta/src/barrier/mod.rs:885]: The original `cancel_stashed_command` only modifies the `finished_jobs`, which is moved to the tracker in this PR. The original logic of `cancel_stashed_command` is merged with the original `find_cancelled_command` into `cancel_command`.

**Review by kwannoel — APPROVED** [review]
LGTM.

**Review by yezizp2012 — APPROVED** [review]
LGTM

**Pull request merged**